lib/galaxy/util/__init__.py

```diff
@@ -294,15 +294,25 @@ def unique_id(KEY_SIZE=128):
     return md5(random_bits).hexdigest()
 
 
-def parse_xml(fname: StrPath, strip_whitespace=True, remove_comments=True) -> ElementTree:
+def parse_xml(
+    fname: StrPath, schemafname: Union[StrPath, None] = None, strip_whitespace=True, remove_comments=True
+) -> ElementTree:
     """Returns a parsed xml tree"""
     parser = None
     if remove_comments and LXML_AVAILABLE:
         # If using stdlib etree comments are always removed,
         # but lxml doesn't do this by default
         parser = etree.XMLParser(remove_comments=remove_comments)
     try:
+        schema = None
+        if schemafname:
+            with open(str(schemafname), "rb") as schema_file:
+                schema_root = etree.XML(schema_file.read())
+                schema = etree.XMLSchema(schema_root)
+
         tree = etree.parse(str(fname), parser=parser)
+        if schema:
+            schema.assertValid(tree)
```

Comment:
Can you do this at the end so the remaining processing occurs, and lxml is not a hard requirements, so this should only be attempted if `LXML_AVAILABLE`

Reply:
Should all be resolved now.